src/twisted/application/runner/_runner.py

```diff
@@ -31,19 +29,73 @@ class Runner(object):
     log = Logger()
 
 
-    def __init__(self, options):
+    def __init__(
+        self,
+        reactor=None,
+        pidFile=nonePIDFile, kill=False,
+        defaultLogLevel=LogLevel.info,
+        logFile=stderr, fileLogObserverFactory=textFileLogObserver,
+        whenRunning=lambda **_: None, whenRunningArguments={},
+        reactorExited=lambda **_: None, reactorExitedArguments={},
+    ):
         """
-        @param options: Configuration options for this runner.
-        @type options: mapping of L{RunnerOptions} to values
+        @param reactor: The reactor to start and run the application in.
+        @type reactor: L{IReactorCore}
```

Comment:
The "incorrect" part of that strikes me as a weak argument; there's a million places where we document a type where `None` is also an option, and I think that requiring that all such cases add an "or None" qualifier in text is super lame.

I also don't agree that forcing the caller to add some boilerplate to import and pass in the default reactor is an improvement here.  The point of this API to to "start the Twisted crap" up for you, not make you figure out how to do that yourself; if you don't have an opinion about which reactor is used, it's OK for this to choose one for you.